### PR TITLE
refactor: dedupe web/mobile scrub-position derivation and shrink ChapterMap (#1351)

### PR DESCRIPTION
**Pull request by seamus-sloan**
## Summary
- Closes #1351. Extracts the "preview while dragging, commit on release" derivation shared by the web and mobile audiobook scrubbers into one `effective_scrub_position()` in `frontend/src/pages/listen/helpers.rs` (a module compiled on both mobile and non-mobile targets), so the two players' bool+f64 / `Option<f64>` signal shapes both funnel through the same math instead of hand-duplicating it.
- Shrinks `ChapterMap` in `frontend/src/pages/listen/chapter_map.rs` from ~138 lines to 94 by extracting the scrub-state derivation into a `scrub_state()` helper and the segmented-bar markup into a `segment_bar()` helper.
- Trims `chapter_map.rs`'s module `//!` doc from 8 lines to 6 (5 content lines), per the ~5-line convention, pointing to the new shared helper for the drag-preview contract detail.
- Mostly a behavior-preserving refactor, with **one small, deliberate exception** flagged by review (see Notes): mobile's drag target is now clamped to `[0, duration]`, matching web's pre-existing clamp, instead of the prior unclamped mobile behavior.

## Test plan
- [x] `cargo fmt -p omnibus-frontend -- --check` — PASS
- [x] `CARGO_TARGET_DIR=/tmp/omnibus-target-1351 cargo clippy -p omnibus-frontend --all-targets --features server -- -D warnings` — PASS (0 warnings)
- [x] `CARGO_TARGET_DIR=/tmp/omnibus-target-1351 cargo test -p omnibus-frontend --features server` — PASS (478 passed; 0 failed), including 6 direct unit tests on `effective_scrub_position`: not-scrubbing passthrough, scrubbing-active override, a chapter-boundary transition, an overshoot-past-the-end clamp, and a negative/underflow clamp.
- [ ] wasm32 clippy (`--target wasm32-unknown-unknown --features web`) — skipped, target not installed in this environment; no wasm-specific code was touched (the new helper is plain `f64`/`usize` arithmetic with no `web`/`wasm` cfg gates).

## Version
- [x] **Patch** — the default; internal refactor plus one small, intentional edge-case behavior alignment (see Notes), no user-facing feature change.

## Notes
- The pre-existing duplication of `chapter_index_for_elapsed` itself (one copy in `chapter_nav.rs` gated `not(mobile)`, one in `mobile/view.rs` gated `mobile`) is left as-is — the issue's shared-helper ask was specifically the "effective position while dragging" derivation, and the two `chapter_index_for_elapsed` copies are threaded into the new shared helper as an `impl Fn` parameter rather than merged, to avoid a larger cross-cfg restructuring outside this issue's scope.
- **Behavior note (from Copilot review):** the shared `effective_scrub_position()` clamps the drag target to `[0, duration]`. Web already did this pre-refactor; mobile's prior derivation used the raw scrub value unclamped (`scrub().unwrap_or(elapsed)`). This is now a deliberate, small behavior fix — aligning mobile with web's existing, safer edge-case handling for a stale/out-of-range drag value (e.g. one that outlives a mid-drag `duration` change) — documented in a comment on `effective_scrub_position` and covered by a new dedicated test (`effective_scrub_position_clamps_a_negative_drag_target_to_zero`).

### CODE REVIEW

File: frontend/src/pages/listen/helpers.rs [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
 
+/// Effective (position, chapter index, remaining) derivation shared by the
+/// web and mobile players' scrub bars: while dragging (`scrub` is `Some`)
+/// every readout previews the target position instead of the real playback
+/// state; at rest (`None`) `elapsed` / `current_chapter_index` / `remaining`
+/// pass through unchanged. `chapter_index_fn` is threaded in rather than
+/// called directly because each target keeps its own
+/// `chapter_index_for_elapsed` under its own cfg gate (`chapter_nav` on web,
+/// `mobile::view` on mobile) — this is the one place the "preview while
+/// dragging, commit on release" math itself lives, so the two can't diverge.
```
[the remark this answers is not in the document]
Comment (seamus-sloan) [reply frontend/src/pages/listen/helpers.rs:25]: Good catch — you're right, this is a real (if small) behavior change, not pure dedup.

I checked both pre-refactor implementations:
- **Web** (`chapter_map.rs`): `scrub_secs().clamp(0.0, max)` — always clamped.
- **Mobile** (`mobile.rs`): `scrub().unwrap_or(elapsed)` — raw, unclamped.

I judged clamping to be the more correct behavior to keep (rather than reproducing mobile's unguarded case): an out-of-range drag value — e.g. one that outlives a mid-drag `duration` change — is an edge case either way, and web's existing clamp is the safer of the two prior behaviors (it prevents a negative or past-the-end `effective` from ever reaching the UI). So I kept the clamp and:

1. Added an explicit note on `effective_scrub_position` in `frontend/src/pages/listen/helpers.rs` calling out that this intentionally aligns mobile's edge-case handling with web's pre-existing clamp, rather than preserving mobile's unguarded one.
2. Added a dedicated test, `effective_scrub_position_clamps_a_negative_drag_target_to_zero`, alongside the existing overshoot-clamp test, so both bounds are covered.
3. Updated the PR description to stop claiming "behavior-preserving refactor only" and call this out as a deliberate, small behavior fix.

Pushed in 86402cc.

---
_Generated by [Claude Code](https://claude.ai/code)_
